september/src/main/java/edu/asu/diging/gilesecosystem/september/web/HomeController.java

```diff
@@ -8,27 +8,60 @@
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
 
+import edu.asu.diging.gilesecosystem.september.core.db.IMessageDbClient;
 import edu.asu.diging.gilesecosystem.september.core.model.IMessage;
+import edu.asu.diging.gilesecosystem.september.core.model.impl.DataTableData;
 import edu.asu.diging.gilesecosystem.september.core.service.IMessageManager;
 
 @Controller
 public class HomeController {
-    
+
     @Autowired
     private IMessageManager manager;
 
+    @Autowired
+    private IMessageDbClient client;
+
     @RequestMapping(value = "/")
     public String home(Principal principal, Model model, @RequestParam(defaultValue = "0") int page) {
         if (principal == null || principal.getName() == null || principal.getName().equals("anonymousUser")) {
             return "home";
         }
-        
+
         List<IMessage> messages = manager.getMessages(page);
         model.addAttribute("messages", messages);
         model.addAttribute("totalPages", manager.getNumberOfPages());
         model.addAttribute("currentPageValue", page);
-        
+        System.out.println("Messages Size: " + messages.size());
         return "home";
     }
+
+    @RequestMapping(value = "/datatable")
+    public @ResponseBody DataTableData doGet(@RequestParam int draw, @RequestParam int start, @RequestParam String type)
+            throws Exception {
+        DataTableData dataTableData = new DataTableData();
+        List<IMessage> dataTableMessages = null;
+        int totalRecords = 0;
+        int filteredRecords = 0;
+        int offset = start / 10;
```

Comment:
why is that divided by 10?

Reply:
Paging first record indicator. This is the start point in the current data set (0 index based - i.e. 0 is the first record). ( https://datatables.net/manual/server-side ). Here offset is literally the page number. So I had to do (start/10) in-order to provide proper offset values (eg: starting from 0 and then 10/10 = 1, 20/10 = 2 and so on).

Reply:
10 is an configurable value, if the app will be configured to use a page size of 20, size will fail/generate wrong results.